homework/src/App.js

```diff
@@ -12,18 +12,25 @@ class App {
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
   async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
     const root = document.getElementById('root');
-    // ...
-
+    const header = Util.createAndAppend('header', root, { 'class': 'header' });
+    Util.createAndAppend('div', root, { 'id': 'container' });
+    Util.createAndAppend('p', header, { 'text': 'HYF Repositories' });
+    const select = Util.createAndAppend('select', header, { 'id': 'select-list', "aria-label": "HYF Repositories" });
     try {
-      // ...
       const repos = await Util.fetchJSON(url);
-      this.repos = repos.map(repo => new Repository(repo));
-      // ...
+      this.repos = repos
+        .sort((x, y) => x.name.localeCompare(y.name))
+        .map(repo => new Repository(repo));
+      repos.forEach((repo, i) => {
```

Comment:
A casual reader might think that only `this.repos` is sorted and that `repos` itself is unmodified. It is only because the `.sort()` array method does an _in-place_ sort that this code works, despite it being used inside a method chain as is done here. You could change this code to the snippet below to take away this confusion:

```js
this.repos.forEach((repo, i) => {
  Util.createAndAppend('option', select, { 'text': repo.name(), 'value': i });
});
```